### PR TITLE
Ce/lxml restore

### PR DESCRIPTION
**Pull request by calellowitz**
##### SUMMARY

This swaps out python core `ElementTree` for `lxml` in restore case and fixture generators. In my profiling of large restores, nearly half the time was spent converting case nodes to a string (one at a time), this cut the time spent doing that by nearly 90%.

The other changes are due to slight differences in their APIs. In lxml, `attrib` is a property and not assignable.

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/restore.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ def append(self, xml_element):
             self.num_items += num - 1
             self.response_body.write(xml_element)
         else:
-            self.response_body.write(xml_util.tostring(xml_element))
+            self.response_body.write(ElementTree.tostring(xml_element, encoding='utf-8'))
```
Comment (calellowitz) [inline corehq/ex-submodules/casexml/apps/phone/restore.py:127]: This part is the only part i am a little uncomfortable with. I only addressed the case and fixture providers (which both returned bytes). Other parts of the restore returned `Element`s and this  does not change them. That means restore providers still return `Element`s (from `ElementTree`) or bytes, but a few of the byte ones use `lxml` behind the scenes. This is not necessarily a clean global switch to lxml.

Comment (czue) [reply corehq/ex-submodules/casexml/apps/phone/restore.py:127]: Can you say more about what in particular you're uncomfortable with about it?

I think this is one of the best-tested parts of the codebase so I think if the tests pass it should be fine?

**Review by czue — APPROVED** [review]
lgtm. long live lxml!

File: corehq/ex-submodules/casexml/apps/phone/restore.py (hunk 1 above)
Comment (calellowitz) [reply corehq/ex-submodules/casexml/apps/phone/restore.py:127]: I don't love that some parts of the restore use python core `ElementTree` and some now use `lxml`, even if the two don't mix (the lxml portions all convert to string before the code reaches here, at least one `ElementTree` portion does not). I am confident it works, both from my testing and the tests you mentioned, but it seems like a potential area for confusion.

**Review by millerdev — APPROVED** [review]
Very nice!

File: corehq/ex-submodules/casexml/apps/case/fixtures.py [hunk 2]
```diff
@@ -28,7 +28,7 @@ def fixture(self):
                 <date_opened/>
                 <last_modified/>
                 <case_property />
-                <index>
+                <index>corehq/ex-submodules/casexml/apps/case/fixtures.py
```
Comment (czue) [inline corehq/ex-submodules/casexml/apps/case/fixtures.py:31]: looks like this bogey snuck in. 

curious why tests aren't failing....

**Review by czue — CHANGES_REQUESTED** [review]
one thing to fix

File: corehq/ex-submodules/casexml/apps/case/fixtures.py (hunk 2 above)
Comment (calellowitz) [reply corehq/ex-submodules/casexml/apps/case/fixtures.py:31]: Its in the docstring, which is why tests arent failing, but not sure how it got there.

Comment (czue) [reply corehq/ex-submodules/casexml/apps/case/fixtures.py]: oh whoops i should have realized that.

**Review by czue — APPROVED** [review]

**Pull request merged**